### PR TITLE
Update demo to federation 2

### PR DESCRIPTION
**Pull request by patrick91**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

*

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the CONTRIBUTING document.
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: reviews/.pdm.toml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[python]
+path = "/Users/patrickarminio/github/strawberry-graphql/federation-demo/reviews/.venv/bin/python"
```
Comment (patrick91) [inline reviews/.pdm.toml]: remove this

File: books/.pdm.toml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+[python]
+path = "/Users/patrickarminio/github/strawberry-graphql/federation-demo/books/.venv/bin/python"
```
Comment (estyxx) [inline books/.pdm.toml]: 👀

File: gateway/index.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+require("dotenv").config();
+
+const { ApolloServer } = require("apollo-server");
+const { ApolloGateway } = require("@apollo/gateway");
+
+const gateway = new ApolloGateway();
```
Comment (estyxx) [inline gateway/index.js:6]:
```suggestion
const gateway = new ApolloGateway(
 supergraphSdl: new IntrospectAndCompose({
    subgraphs: [
      {
        name: "books",
        url: "http://books:4000/graphql",
      },
      {
        name: "reviews",
        url: "http://reviews:4000/graphql",
      },
    ],
  }),
);
```

Why dotenv? 🤔

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,13 @@
+publish:
+	cd books && pdm run strawberry export-schema app > ../books.graphql
+	cd reviews && pdm run strawberry export-schema app > ../reviews.graphql
+
+	APOLLO_KEY=. \
+		rover subgraph publish strawberry-federation@current \
+		--name reviews --schema ./reviews.graphql \
+		--routing-url http://reviews:4000/graphql
+
+	APOLLO_KEY=. \
```
Comment (estyxx) [inline Makefile]:
```suggestion
include .env

publish:
	cd books && pdm run strawberry export-schema app > ../books.graphql
	cd reviews && pdm run strawberry export-schema app > ../reviews.graphql

	APOLLO_KEY=$(APOLLO_KEY) \
		rover subgraph publish strawberry-federation@current \
		--name reviews --schema ./reviews.graphql \
		--routing-url http://reviews:4000/graphql

	APOLLO_KEY=$(APOLLO_KEY) \
```

Just learned: just putting `include` will do the trick!
Also the geraph name should be a variable maybe?

**Pull request merged**